File: app/src/main/java/com/joshuahalvorson/safeyoutube/adapter/PlaylistItemsListRecyclerviewAdapter.java
package com.joshuahalvorson.safeyoutube.adapter;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.joshuahalvorson.safeyoutube.R;
import com.joshuahalvorson.safeyoutube.model.Item;

import java.util.ArrayList;

public class PlaylistItemsListRecyclerviewAdapter extends RecyclerView.Adapter<PlaylistItemsListRecyclerviewAdapter.ViewHolder>{
    private ArrayList<Item> items;

    public PlaylistItemsListRecyclerviewAdapter(ArrayList<Item> items) {
        this.items = items;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        return new
                ViewHolder(
                        LayoutInflater.from(viewGroup.getContext())
                                .inflate(
                                        R.layout.playlist_items_list_element_layout,
                                        viewGroup,
                                        false
                                )
        );
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {
        Item item = items.get(i);
        viewHolder.videoName.setText(item.getSnippet().getTitle());
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder{
        TextView videoName;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            videoName = itemView.findViewById(R.id.video_name);
        }
    }
}

File: app/src/main/java/com/joshuahalvorson/safeyoutube/Kotlin/adapter/PlaylistItemsListRecyclerviewAdapter.kt
package com.joshuahalvorson.safeyoutube.Kotlin.adapter

import android.support.constraint.ConstraintLayout
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.joshuahalvorson.safeyoutube.Kotlin.model.Models
import com.joshuahalvorson.safeyoutube.R
import java.util.ArrayList

class PlaylistItemsListRecyclerviewAdapter(
        private val items: ArrayList<Models.Item>, private val callback: OnVideoClicked
): RecyclerView.Adapter<PlaylistItemsListRecyclerviewAdapter.ViewHolder>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, p1: Int): PlaylistItemsListRecyclerviewAdapter.ViewHolder {
        return ViewHolder(
                LayoutInflater.from(viewGroup.context)
                        .inflate(R.layout.playlist_items_list_element_layout, viewGroup, false)
        )
    }

    override fun onBindViewHolder(viewHolder: PlaylistItemsListRecyclerviewAdapter.ViewHolder, position: Int) {
        val item = items[position]
        val name = item.snippet?.title
        viewHolder.videoName.setText(name)
        Glide.with(viewHolder.videoThumbnail.getContext())
                .load(item.snippet?.thumbnails?.standard?.url)
                .into(viewHolder.videoThumbnail)

        viewHolder.videoParent.setOnClickListener(View.OnClickListener { callback.onVideoClicked(position) })
    }

    override fun getItemCount(): Int {
        return items.size
    }

    class ViewHolder(view: View): RecyclerView.ViewHolder(view){
        val videoParent = itemView.findViewById<ConstraintLayout>(R.id.video_parent)
        val videoName = itemView.findViewById<TextView>(R.id.video_name)
        val videoThumbnail = itemView.findViewById<ImageView>(R.id.video_thumbnail)
    }

    interface OnVideoClicked {
        fun onVideoClicked(itemIndex: Int)
    }

}